bluesky/scans.py

```diff
@@ -64,18 +68,34 @@ def __iter__(self):
         for flyer in self.flyers:
             yield Msg('collect', flyer, block_group='_flyers')
             yield Msg('wait', None, '_flyers')
-        yield from self._post_scan()
         yield Msg('close_run')
+        yield from self._post_scan()
 
     def _pre_scan(self):
-        for obj in self.objects:
-            conf = self.configuration.get(obj, {})
-            yield Msg('configure', obj, conf)
+        """
+        Subclasses use this to inject some messages to be processed
+        before the run is opened -- for example, configuration or
+        taking some preliminary readings needed for metadata.
+        """
+        # prerun is expected to be a callable that takes the Scan object
+        # itself as its argument and returns a generator of messages
+        if self.pre_run is not None:
+            yield from self.pre_run(self)
```

Comment:
I think the point is 'stash them if you want to and how you want to'. The original suggestion of a tuple `(fcn, args, kwargs)` for both pre and post-run is annoying to deal with (in my opinion).